packages/twenty-front/src/modules/object-record/record-board/record-board-column/components/RecordBoardColumnNewOpportunity.tsx

```diff
@@ -55,22 +69,24 @@ export const RecordBoardColumnNewOpportunity = ({
     openRightDrawer(RightDrawerPages.ViewRecord);
 
     if (isDefined(createdCompany)) {
-      handleEntitySelect(position, createdCompany);
+      createNewBoardRecord(columnId, position);
     }
   };
 
   return (
     <>
-      {newRecord.isCreating && newRecord.position === position && (
+      {shouldDisplayCompanyPicker && (
         <OverlayContainer>
           <RecordPickerComponentInstanceContext.Provider
             value={{ instanceId: RelationPickerHotkeyScope.RelationPicker }}
           >
             <SingleRecordSelect
-              onCancel={() => handleCreateSuccess(position, columnId, false)}
-              onRecordSelected={(company) =>
-                company ? handleEntitySelect(position, company) : null
-              }
+              onCancel={() => createNewBoardRecord(columnId, position)}
+              onRecordSelected={(company) => {
+                if (isDefined(company)) {
+                  upsertBoardOpportunityRecord(columnId, company);
```

Comment:
so I would not upsert here but createNewBoardRecord + write in recordStore instead.